src/consensus/pbft_manager.cpp

```diff
@@ -1332,31 +1333,22 @@ void PbftManager::pushSyncedPbftBlocksIntoChain_() {
                                                      sortition_threshold_, TWO_T_PLUS_ONE)) {
       // Failed cert votes validation, flush synced PBFT queue and set since
       // next block validation depends on the current one
-      LOG(log_er_) << "Synced PBFT block " << pbft_block_and_votes.pbft_blk->getBlockHash()
-                   << " doesn't have enough valid cert votes. Clear synced PBFT blocks!"
-                   << " DPOS total votes count: " << getDposTotalVotesCount();
+      LOG(log_er_) << "Synced PBFT block " << pbft_block_hash
+                   << " doesn't have enough valid cert votes. Clear synced PBFT blocks! DPOS total votes count: "
+                   << getDposTotalVotesCount();
       pbft_chain_->clearSyncedPbftBlocks();
       break;
     }
-    if (!pbft_chain_->checkPbftBlockValidation(*pbft_block_and_votes.pbft_blk)) {
-      // PBFT chain syncing faster than DAG syncing, wait!
-      pbft_synced_queue_size = pbft_chain_->pbftSyncedQueueSize();
-      if (pbft_last_observed_synced_queue_size_ != pbft_synced_queue_size) {
-        LOG(log_dg_) << "PBFT chain unable to push synced block " << pbft_block_and_votes.pbft_blk->getBlockHash();
-        LOG(log_dg_) << "PBFT synced queue still contains " << pbft_synced_queue_size
-                     << " synced blocks that could not be pushed.";
-      }
-      pbft_last_observed_synced_queue_size_ = pbft_synced_queue_size;
-      break;
-    }
+
     if (!comparePbftBlockScheduleWithDAGblocks_(*pbft_block_and_votes.pbft_blk)) {
+      LOG(log_nf_) << "DAG has not build up for anchor " << pbft_block_and_votes.pbft_blk->getPivotDagBlockHash()
+                   << " in PBFT block " << pbft_block_hash;
       break;
     }
     if (pushPbftBlock_(pbft_block_and_votes)) {
```

Comment:
Here is the real perf problem. We call getDagBlockOrder inside "pushPbftBlock_" as well as in  "comparePbftBlockScheduleWithDAGblocks_", which is called few lines above...

Reply:
Thanks, fixed